packages/fxa-shared/experiments/base.ts

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import md5 = require('js-md5');
+
+class BaseGroupingRule {
+  private name: string = '';
+  private deprecated: boolean = false;
+
+  /**
+   * Return a 32 bit hash of `key`
+   *
+   * @param {String} key
+   * @returns {Number}
+   */
+  public hash(key: string): number {
+    // md5 returns 32 hex bytes, we want 32 bits. 32bits = 8 hex bytes.
+    const hash = md5(`${this.name || ''}:${key}`).substr(0, 8);
```

Comment:
nit: since the `name` member has a default of empty string, `|| ''` isn't necessary here.